### PR TITLE
Rails 7 defaults

### PR DESCRIPTION
**Pull request by parsahonarmand99**
Upgrades this application to use Rails 7.0 defaults.

Upgraded `config.load_defaults` in `application.rb` followed by `bin/rails app:update`. 

Ensured the new defaults are compatible with this Rails app.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@ def authenticate_with_identity
       secure: true,
     }
 
-    redirect_to identity_authenticate_url(identity_state)
+    redirect_to identity_authenticate_url(identity_state), allow_other_host: true
```
Comment (parsahonarmand99) [inline app/controllers/application_controller.rb:23]: Rails 7 prevents redirects to external hosts by default. https://api.rubyonrails.org/v7.0.8.4/classes/ActionController/Redirecting.html#method-i-redirect_to

File: config/initializers/filter_parameter_logging.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Be sure to restart your server when you modify this file.
 
-# Configure sensitive parameters which will be filtered from the log file.
+# Configure parameters to be filtered from the log file. Use this to limit dissemination of
+# sensitive information. See the ActiveSupport::ParameterFilter documentation for supported
+# notations and behaviors.
 Rails.application.config.filter_parameters += [
-  :passw, :password, :secret, :token, :_key, :crypt, :salt, :certificate, :otp, :ssn
```
Comment (arkalim) [inline config/initializers/filter_parameter_logging.rb:5]: I think it's best to have both `passw` and `:password` present here.

**Review by arkalim — COMMENTED** [review]
Just a small comment. The rest looks good.

**Review by arkalim — APPROVED** [review]

File: config/initializers/filter_parameter_logging.rb (hunk 2 above)
Comment (parsahonarmand99) [reply config/initializers/filter_parameter_logging.rb:5]: Ah good catch! Fixed

**Review by anthonyoconnorclio — APPROVED** [review]

**Pull request merged**